### DISCUSSION TITLE
Target Audience for Search verticals and Filter web part

### DISCUSSION THREAD

**Discussion by Viipiin** [microsoft-search/pnp-modern-search#2052] [Ideas]
Version 4.0
**Is your feature request related to a problem? Please describe.**
I am frustrated when I want to show tabs (search verticals) based on permission. I have 2 tabs and 2 search results web parts on my page but I want to show both tabs to specific users based on a group. I want to hide one of the vertical for many users based on their roles.

**Describe the solution you'd like**
It would be great if we can have a target audience feature for Search Verticals and Search filter web parts.

**Comment by edarroudi** [comment]
This shouldn't be too complex to implement. Maybe something like this:
![image](https://user-images.githubusercontent.com/935493/165269448-88e13d77-336e-4c5b-8823-c2b21360e8e1.png)

Maybe some issues with performace (checking if user is member of group...) 

I'll think about it. I would feel better to implement this as an interface for each Webpart, Something like Audience Targeting Settings for all modern search webparts. In case of Verticals on an "vertical entry level".

**Comment by wobba** [comment]
A bit related to https://github.com/microsoft-search/pnp-modern-search/discussions/2047 but that's about hiding the full web part.
